### PR TITLE
Add missing `NotImplementedError` for `VectorEnv.step` 

### PR DESCRIPTION
**Pull request by jjshoots**
# Description

Fixes some broken typehints.

### CODE REVIEW

File: gymnasium/vector/vector_env.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ class VectorEnv(Generic[ObsType, ActType, ArrayType]):
     def reset(
         self,
         *,
-        seed: int | None = None,
+        seed: int | list[int] | None = None,
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/vector/vector_env.py:130]: This was an intentional decision that is overwritten by the `AsyncVectorEnv` and `SyncVectorEnv` 
but not for `VectorEnv`

File: gymnasium/vector/vector_env.py [hunk 2]
```diff
@@ -190,6 +192,7 @@ def step(
             >>> infos
             {}
         """
+        raise NotImplementedError(f"{self.__str__()} step function is not implemented.")
```
Comment (pseudo-rnd-thoughts) [inline gymnasium/vector/vector_env.py:195]: This does seem helpful to include

Comment (jjshoots) [reply gymnasium/vector/vector_env.py:195]: I'm not sure how the formatter didn't catch this function without a body. Is that normal?

File: gymnasium/vector/vector_env.py (hunk 1 above)
Comment (jjshoots) [reply gymnasium/vector/vector_env.py:130]: Ah gotcha, will revert.

File: gymnasium/vector/vector_env.py (hunk 2 above)
Comment (pseudo-rnd-thoughts) [reply gymnasium/vector/vector_env.py:195]: It is because the docstring counts as a function component

File: gymnasium/vector/vector_env.py (hunk 1 above)
Comment (pseudo-rnd-thoughts) [reply gymnasium/vector/vector_env.py:130]: Thanks

**Review by pseudo-rnd-thoughts — APPROVED** [review]

**Pull request merged**